python/pyarrow/tests/test_dataset.py

```diff
@@ -2200,7 +2196,7 @@ def test_write_dataset(tempdir):
     dataset = ds.dataset(directory)
 
     target = tempdir / 'single-directory-target'
-    expected_files = [target / "dat_0.ipc", target / "dat_1.ipc"]
+    expected_files = [target / "part-0.feather"]
```

Comment:
Why did this change to a single file? (the original has 2 files, I expect the roundtrip to preserve those files)

Reply:
After this patch, a single file will be written for each partition directory. In a follow up we'll add an optional cap on file size